File: VestTour/Entities/Voucher.cs
using System;
using System.Collections.Generic;

namespace VestTour.Models;

public partial class Voucher
{
    public int VoucherId { get; set; }

    public string? Status { get; set; }

    public string? VoucherCode { get; set; }

    public string? Description { get; set; }

    public decimal? DiscountNumber { get; set; }

    public DateOnly? DateStart { get; set; }

    public DateOnly? DateEnd { get; set; }

    public virtual ICollection<Booking> Bookings { get; set; } = new List<Booking>();

    public virtual ICollection<Order> Orders { get; set; } = new List<Order>();
}

File: Mobile/mobileapp/lib/models/voucher.dart
class Voucher {
  int? voucherId;
  String? status;
  String? voucherCode;
  String? description;
  double? discountNumber;
  String? dateStart;
  String? dateEnd;

  Voucher(
      {this.voucherId,
        this.status,
        this.voucherCode,
        this.description,
        this.discountNumber,
        this.dateStart,
        this.dateEnd});

  Voucher.fromJson(Map<String, dynamic> json) {
    voucherId = json['voucherId'];
    status = json['status'];
    voucherCode = json['voucherCode'];
    description = json['description'];
    discountNumber = json['discountNumber'];
    dateStart = json['dateStart'];
    dateEnd = json['dateEnd'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['voucherId'] = this.voucherId;
    data['status'] = this.status;
    data['voucherCode'] = this.voucherCode;
    data['description'] = this.description;
    data['discountNumber'] = this.discountNumber;
    data['dateStart'] = this.dateStart;
    data['dateEnd'] = this.dateEnd;
    return data;
  }
}
